src/modules/socket.ts

```diff
@@ -2,147 +2,225 @@ import { io, Socket } from "socket.io-client";
-  pause: boolean;
-  play: boolean;
-  seek: boolean;
-  chat: boolean;
-  kick: boolean;
+    pause: boolean;
+    play: boolean;
+    seek: boolean;
+    chat: boolean;
+    kick: boolean;
+}
+interface MessageEvent {
+    play: boolean;
+    pause: boolean;
+    seek: boolean;
 }
 
 export class ClientSocketHandler {
-  private serverUrl: string;
-  private socket!: Socket;
-  private video: HTMLVideoElement;
-  private permissions!: Permissions;
+    private serverUrl: string;
+    private socket!: Socket;
+    private video: HTMLVideoElement;
+    private permissions!: Permissions;
+    private messageEvent!: MessageEvent;
+    private seek: boolean = false;
 
-  constructor(video: HTMLVideoElement) {
-    this.serverUrl = "http://localhost:5000";
-    this.video = video;
-  }
+    constructor(video: HTMLVideoElement) {
+        this.serverUrl = "http://localhost:5000";
+        this.video = video;
+        this.messageEvent = { play: false, pause: false, seek: false };
+    }
 
-  openConnection = () => {
-    this.socket = io(this.serverUrl);
-    this.socket.on("connect", () => {
-      console.log("Socket is connected");
-    });
-    this.socket.on("disconnect", () => {
-      console.log("Socket is disconnected");
-    });
-    this.socket.on(
-      SocketEventType.RECEIVE_VIDEO_EVENT,
-      (message: EventInfo) => {
-        console.log("Received video event from the server: ", message);
-        switch (message.event) {
-          case "play":
-            this.video.play();
-            console.log("Video is played!");
-            break;
-          case "pause":
-            this.video.pause();
-            console.log("Video is paused!");
-            break;
-          case "seeked":
-            //this.video.currentTime = message.currentTime;
-            console.log("Video is seeked!", message.currentTime);
-            break;
-        }
-      }
-    );
-    this.socket.on(SocketEventType.PERMISSIONS, (message: any) => {
-      console.log("Received video event from the server: ", message);
-      this.permissions = message.permissions;
-    });
-  };
+    openConnection = () => {
+        this.socket = io(this.serverUrl);
+        this.socket.on("connect", () => {
+            console.log("Socket is connected");
+        });
+        this.socket.on("disconnect", () => {
+            console.log("Socket is disconnected");
+        });
+        this.socket.on(
+            SocketEventType.RECEIVE_VIDEO_EVENT,
+            (message: EventInfo) => {
+                this.seek = true;
+                console.log("Received video event from the server: ", message);
+                switch (message.event) {
+                    case "play":
+                        if (Math.abs(this.video.currentTime - message.currentTime) > 0.5 ) {
+                            const time1 = message.currentTime * 1000;
+                            const seekCode1 = `const videoPlayer = netflix.appContext.state.playerApp.getAPI().videoPlayer;
```

Comment:
This is the most important part of the change, when we receive the event, we use a netflix API call to control the video, if normal HTML function would be used, then the player would break.